api/src/main/java/org/apache/iceberg/expressions/UnboundPredicate.java

```diff
@@ -125,13 +154,38 @@ public Expression bind(Types.StructType struct, boolean caseSensitive) {
         case LT_EQ:
         case EQ:
           return Expressions.alwaysFalse();
-//        case IN:
-//          break;
-//        case NOT_IN:
-//          break;
       }
     }
     return new BoundPredicate<>(op(), new BoundReference<>(field.fieldId(),
-        schema.accessorForField(field.fieldId())), lit);
+            schema.accessorForField(field.fieldId())), lit);
+  }
+
+  @SuppressWarnings("unchecked")
+  private Expression bindInOperation(Types.NestedField field, Schema schema) {
+    final Set<Literal<T>> lits = literalSet().stream().map(
+        val -> {
+          if (val instanceof LiteralSet.CharSeqWrapper) {
+            val = (T) ((LiteralSet.CharSeqWrapper) val).unWrap();
+          }
+          Literal<T> lit = Literals.from(val).to(field.type());
```

Comment:
I don't think this should reconstruct literals. Instead, I think `UnboundPredicate` should have a `List<Literal<T>> literals` that gets transformed. That would work for all operations.

Reply:
Sure, I will change it.